### DISCUSSION TITLE
Create a PSA app with WindowsAPP SDK

### DISCUSSION THREAD

**Discussion by nitish14s** [microsoft/WindowsAppSDK#4301] [Dev team discussions]
I am trying to create a PSA app using WindowsAPPSDK (WinUI3). I have created it as a multiinstancing app and updated the Package.AppManifest of the app 
<Extensions>
 <printsupport:Extension Category="windows.printSupportSettingsUI" EntryPoint="$targetentrypoint$"/>
</Extensions>

But when while opening from a IPP enabled printer to open this app on clicking the Printer Preferences, app is crashing with 
System.Runtime.InteropServices.COMException

**Comment by jonwis** [comment]
@mziller can you assist or route?

@nitish14s can you file a Feedback Hub item and use the "record" option to collect a trace of how your app is being launched? What is the callstack reported by VS of the COMException? (Feel free to anonymize your DLL names of course.)

**Reply by nitish14s** [reply]
Posted as an issue. Please follow up on https://github.com/microsoft/WindowsAppSDK/issues/4304

**Comment by mziller** [comment]
Hardware Support Apps are intended for hardware vendors to be able to extend functionality of their devices. They're not a general extensibility mechanism for end users to add customizations. Printer vendors should reach out to Microsoft directly for assistance in developing PSAs

**Reply by dev-nike** [reply]
I am a developer working for a printer manufacturer. Could you please provide a contact method through which we can obtain support for PSA (Print Support App) development?

**Reply by lauren-ciha** [reply]
Developers can connect with Microsoft through the [Microsoft Partner Center](https://partner.microsoft.com/en-US/). To register your organization, follow the [enroll link here](https://partner.microsoft.com/en-us/dashboard/account/v3/enrollment/introduction/partnership).
